File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        ans= []
        hashmap = {}
        for i in range(len(nums)):
            required =target - nums[i]
            if required in hashmap:
                return [i,hashmap[required]]
            hashmap[nums[i]] = i
        

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int [] ans = new int[2];
        HashMap<Integer,Integer> mp = new HashMap<>();
        for (int i = 0 ; i < nums.length ; i++){
            int curr = nums[i];
            int remaining  =  target - curr;
            if (mp.containsKey(remaining)){
                ans[0] = i;
                ans[1] = mp.get(remaining);
                return ans;
            }
            mp.put(nums[i],i);
            
        }
        // for (Map.Entry <Integer,Integer> ele = mp.getSet();)
        return ans;
    }
}